memtx: fix a crash caused by mhash misusage in MVCC (#11162)
tarantool/tarantool: changelogs/unreleased/gh-11022-memtx-mvcc-point-holes-mhash-abort.md

## bugfix/memtx

> * Fixed the issue when Tarantool with memtx MVCC enabled was aborted on

Reviewer: ```suggestion
* Fixed a bug when Tarantool with memtx MVCC enabled was aborted on
```
